keps/sig-instrumentation/4785-resource-state-metrics/README.md

```diff
@@ -0,0 +1,1170 @@
+
+For each of these, fill in the following—thinking about running existing user workloads
+and creating new ones, as well as about cluster-level services (e.g. DNS):
+  - [Dependency name]
+    - Usage description:
+      - Impact of its outage on the feature:
+      - Impact of its degraded performance or high-error rates on the feature:
+-->
+
+The controller relies on core Kubernetes components.
+
+### Scalability
+
+<!--
+For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them.
+
+For beta, this section is required: reviewers must answer these questions.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### Will enabling / using this feature result in any new API calls?
+
+<!--
+Describe them, providing:
+  - API call type (e.g. PATCH pods)
+  - estimated throughput
+  - originating component(s) (e.g. Kubelet, Feature-X-controller)
+Focusing mostly on:
+  - components listing and/or watching resources they didn't before
+  - API calls that may be triggered by changes of some Kubernetes resources
+    (e.g. update of object X triggers new updates of object Y)
+  - periodic API calls to reconcile state (e.g. periodic fetching state,
+    heartbeats, leader election, etc.)
+-->
+
+Yes, the controller needs to make API calls in order to reconcile the managed
+resources in the cluster. There are no resync polls done by the controller
+(`ResyncPeriod: 0` for all reflectors). The reflectors will do a
+LIST and/or WATCH on the associated resources' modification. The same applies
+for managed resource(s), however, their case will be accompanied by additional
+GET and UPDATE calls to ensure their `ObjectMeta` and `Status` are synced.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+<!--
+Describe them, providing:
+  - API type
+  - Supported number of objects per cluster
+  - Supported number of objects per namespace (for namespace-scoped objects)
+-->
+
+Yes, the controller currently has one cluster-scoped managed resource of  
+[`ResourceMetricsMonitor` type]. There is currently no upper-limit on the number of
+managed resource instances that could be defined.
+
+[`ResourceMetricsMonitor` type]: https://github.com/rexagod/resource-state-metrics/blob/main/pkg/apis/resourcestatemetrics/v1alpha1/types.go
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+<!--
+Describe them, providing:
+  - Which API(s):
+  - Estimated increase:
+-->
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+<!--
+Describe them, providing:
+  - API type(s):
+  - Estimated increase in size: (e.g., new annotation of size 32B)
+  - Estimated amount of new objects: (e.g., new Object X for every existing Pod)
+-->
+
+The controller will **not** create or modify any object on its own. Only when a
```

Comment:
This is a bit misleading - it actually is modifying managed resources when it's created/updated.